src/app-layout/drawer/index.tsx

```diff
@@ -131,47 +134,136 @@ export const Drawer = React.forwardRef(
+  const isCompactMode = useCompactMode();
+
+  const getIndexOfOverflowItem = () => {
+    if (containerHeight) {
+      const ITEM_HEIGHT = isCompactMode ? 34 : 38;
+      const overflowSpot = containerHeight / 1.5;
+
+      const index = Math.floor(overflowSpot / ITEM_HEIGHT);
+
+      return index;
+    }
+    return 0;
+  };
+
+  const splitItems = (items: any[] | undefined) => {
+    let visibleItems = [];
+    let overflowItems = [];
+    if (items) {
+      visibleItems = items.slice(0, getIndexOfOverflowItem());
+      overflowItems = items.slice(getIndexOfOverflowItem(), items.length);
+    }
+    return { visibleItems, overflowItems };
+  };
+
+  const overflowItemHasBadge = () => {
+    const { overflowItems } = splitItems(drawers?.items);
+    return overflowItems ? overflowItems.filter(item => item.badge).length > 0 : false;
+  };
+
+  const overflowItemIsActive = () => {
+    const { overflowItems } = splitItems(drawers?.items);
+
+    if (drawers && drawers.activeDrawerId) {
+      return overflowItems.map(item => item.id).indexOf(drawers.activeDrawerId) !== -1;
+    }
+    return false;
+  };
+
+  const getDrawerItems = () => {
+    if (drawers && drawers.items) {
+      const activeIndex = drawers.activeDrawerId && drawers.items.map(item => item.id).indexOf(drawers.activeDrawerId);
+      const lastMainItemIndex = getIndexOfOverflowItem() - 1;
+
+      const activeDrawerItem = drawers.items.find(item => item.id === drawers.activeDrawerId);
+      const lastMainItem = drawers.items.find((item, index) => index === lastMainItemIndex);
+
+      if (activeIndex && overflowItemIsActive() && activeDrawerItem && lastMainItem) {
+        [drawers.items[lastMainItemIndex], drawers.items[activeIndex]] = [
+          drawers.items[activeIndex],
+          drawers.items[lastMainItemIndex],
+        ];
+      }
+
+      return drawers.items;
+    }
+    return [];
+  };
+
+  const { visibleItems, overflowItems } = splitItems(getDrawerItems());
+
   return (
     <div
       className={clsx(styles.drawer, styles['drawer-closed'], testutilStyles['drawer-closed'], {
         [styles['drawer-mobile']]: isMobile,
       })}
     >
-      <div style={{ top: topOffset, bottom: bottomOffset }} className={styles['drawer-content']}>
+      <div
+        ref={triggersContainerRef}
+        style={{ top: topOffset, bottom: bottomOffset }}
+        className={clsx(styles['drawer-content'])}
+      >
         {!isMobile && (
           <aside aria-label={drawers?.ariaLabel} className={clsx(styles['drawer-triggers-wrapper'], contentClassName)}>
-            {drawers?.items?.map((item: DrawerItem, index: number) => (
-              <span
-                key={index}
-                className={clsx(
-                  styles['drawer-trigger'],
-                  drawers.activeDrawerId === item.id && styles['drawer-trigger-active']
-                )}
-                onClick={() =>
```

Comment:
Why do we need to move the handler? Now there is an issue that the trigger outside blue area is not clickable
<img width="60" alt="image" src="https://github.com/cloudscape-design/components/assets/812240/3f573350-8d49-48c4-aa89-014456ce6f5f">

Reply:
The handler was being called twice, once on the span and once on the Toggle button. I needed to remove one of them. I can remove the one on the ToggleButton instead